src/main/java/com/amazon/opendistroforelasticsearch/sql/optimizer/subquery/model/SubqueryType.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.optimizer.subquery.model;
+
+public enum SubqueryType {
```

Comment:
I am thinking if it is better to put this in the `Subquery` class as a static nested type: `Subquery.Type`